mtags/src/main/scala/scala/meta/internal/mtags/MtagsEnrichments.scala

```diff
@@ -359,4 +363,31 @@ trait MtagsEnrichments {
     def isVar: Boolean = (properties & p.VAR.value) != 0
     def isVal: Boolean = (properties & p.VAL.value) != 0
   }
+
+  implicit class XtensionStream[A](stream: java.util.stream.Stream[A]) {
+    import scala.collection.JavaConverters._
+    def asScala: Generator[A] = {
+      Generator.selfClosing((stream.iterator.asScala, () => stream.close()))
+    }
+  }
+
+  implicit class XtensionAbsolutePath(path: AbsolutePath) {
+    def parent: AbsolutePath = {
+      AbsolutePath(path.toNIO.getParent)
+    }
+
+    def exists: Boolean = {
+      Files.exists(path.toNIO)
+    }
+
+    def walk: Generator[AbsolutePath] = {
+      if (exists) Files.walk(path.toNIO).asScala.map(AbsolutePath(_))
```

Comment:
Looks like there's an implicit conversion going on here and the stream won't be closed. I would use the `Geny.selfClosing` constructor to ensure that it's properly closed.

Reply:
streams don't have an implicit conversion with `asScala` - that is [my own addition](https://github.com/scalameta/metals/pull/938/files/392c3f84dd8bc06a82310a7f14b0358bfd1c9465#diff-ef3c89457c7ae83653bcdf6c20438e96R369). It tries to mimick the scala conversion to avoid any surprises on the callee site (like: "what was the name of that special method for streams?") - I want it to be as seamless as possible.

Reply:
Aha, I'm sorry I didn't see that one. Looks great 👍